src/scos_tekrsa/hardware/tekrsa_sigan.py

```diff
@@ -259,19 +285,13 @@ def preamp_enable(self, preamp_enable):
     def healthy(self, num_samples=56000):
         """Perform health check by collecting IQ samples."""
         logger.debug("Performing Tektronix RSA health check.")
-
         try:
-            measurement_result = self.acquire_time_domain_samples(num_samples)
-            data = measurement_result["data"]
+            return not self.rsa.DEVICE_GetOverTemperatureStatus()
```

Comment:
Why the change in functionality here?

Regardless, the `healthy` method needs its docstring updated to reflect the change, and we can get rid of the `num_samples` parameter since it is no longer used.

Reply:
I believe healthy could be called while the signal analyzer is executing a measurement action and wasn't sure how the signal analyzer would respond. Even if the underlying signal analyzer is smart enough to block it seemed it could possibly introduce significant delays at the very least and worse if the sigan doesn't block.

Reply:
That makes sense. I think this could still potentially cause problems if run while an action is executing (if any API calls are made at the same time). This most likely would be an issue if the sigan is collecting samples while `healthy` is called, but I don't know how the API would handle this. It could result in `healthy` being `False` just because the signal analyzer was actively collecting samples during the health check, which would be incorrect. Or it could be worse and cause the measurement to fail. We should be careful to avoid running health checks while actions are running, at the very least, without knowing how the signal analyzer responds to this case.